File: python/person.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Person(object):

	nbr = 0

	def __init__(self, name, byear):
		self.name = name
		self.byear = byear
		Person.nbr = Person.nbr + 1

	def __del__(self):
		Person.nbr = Person.nbr - 1
		print (self.name + " is out")

	def info(self):
		print("My name: " + self.name + ", My birth year: " + str(self.byear))

	@staticmethod
	def population():
		return Person.nbr

p = Person("Karim", 1986)
p2 = Person("Karim+1", 1987);

p.info()
p2.info()
nbr = Person.population()

print ("The number of persons: " + str(nbr));
del p2
nbr = Person.population()
print ("The number of persons: " + str(nbr));

File: cpp/redefinition/person.cpp
#include "person.h"

Person::Person(std::string name, int age)
{
	this->name = name;
	this->age = age;
}

void Person::info()
{
	std::cout << "My name: " << this->name << "\n";
	std::cout << "My age: " << this->age << "\n";
}

/*
int main()
{
	Person p = Person("Karim", 31);
    p.info();
} 
*/
